src/org/authInfo.ts

```diff
@@ -392,21 +392,29 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     // authInfo before it is necessary.
     const logger = await Logger.child('Common', { tag: 'identifyPossibleScratchOrgs' });
 
-    // return if we already know the hub org we know it is a devhub or prod-like or no orgId present
+    // return if we already know the hub org, we know it is a devhub or prod-like, or no orgId present
     if (fields.isDevHub || fields.devHubUsername || !fields.orgId) return;
 
-    logger.debug('getting devHubs');
+    logger.debug('getting devHubs and prod orgs to identify scratch orgs and sandboxes');
 
     // TODO: return if url is not sandbox-like to avoid constantly asking about production orgs
     // TODO: someday we make this easier by asking the org if it is a scratch org
 
     const hubAuthInfos = await AuthInfo.getDevHubAuthInfos();
+    // Get a list of org auths that are known not to be devhubs, scratch orgs, or sandboxes.
+    const possibleProdOrgs = await AuthInfo.listAllAuthorizations(
```

Comment:
since it's possible that a DevHub is also a ProdOrg, you could include DevHubs in the filter.

Then instead of having to `queryScratchOrg` each DevHub and then catch to ask about sandbox, you could ask about everything on the first round.